### PR TITLE
feat: convert SignPosts and LanguageList shortcodes to Nunjunks templates

### LINKED ISSUE #9775: Convert non-author shortcodes `{% SignPosts %}` and `{% LanguageList %}` to includes
https://github.com/GoogleChrome/web.dev/pull/9765 surfaced we have a `{% SignPosts %}` shortcode, which is only used from njk templates and only serves layout purposes. This should instead be refactored to a simple njk `{% include %`}.

### PR DESCRIPTION
**Pull request by mamieorine**
Fixes #9775 

Since both `{% SignPosts %}` and `{% LanguageList %}` shortcode which are only used from Nunjunks templates, this should instead be refactored to a Nunjunks template and imported to the content using {% include %}.

Changes proposed in this pull request:

- Remove `{% SignPosts %}` shortcode from the eleventy config
- Remove `{% LanguageList %}` shortcode from the eleventy config
- Convert SignPosts shortcode to a simple a Nunjunks template
- Convert LanguageList shortcode to a Nunjunks template 

For converting the LanguageList shortcode to Nunjunks, this component needs a javascript file that has logic for finding the available language for each post content. Thus, I kept the `LanguageList.js `and imported it into `content.11tydata.js.`. Please let me know if you have any suggestions on this, thank you.

### CODE REVIEW

File: src/site/_includes/codelab.njk [hunk 1]
```diff
@@ -29,7 +29,8 @@ pageScripts:
           </div>
         {% endif %}
 
-        {% SignPosts page.fileSlug %}
+        {% include 'partials/sing-post.njk' %}
```
Comment (matthiasrohmer) [inline src/site/_includes/codelab.njk:32]: New keyboard? 😄  `sign`.

**Review by matthiasrohmer — COMMENTED** [review]
Bunch of typos. My suggestion is to remove the remaining needed logic to a filter instead and then do something like `page | languageList` in the template. Make sure to also move the `LanguageList` file out of the components directory then.

**Comment by mamieorine** [comment]
@matthiasrohmer  I have fixed all typos and changed the way to enable using of functions in the Nunjunks template by implementing the supportedLanguages filter instead.

- Multiple languages of the content available
   https://deploy-preview-9849--web-dev-staging.netlify.app/defining-core-web-vitals-thresholds/

- Only English language available
   https://deploy-preview-9849--web-dev-staging.netlify.app/push-notifications-in-all-modern-browsers/

File: src/site/_filters/language-list.js [hunk 2]
```diff
@@ -15,52 +15,55 @@
...
-const {i18n} = require('../../_filters/i18n');
+  getLocaleSpecificUrl,
+} = require('../_filters/urls');
+
+/**
+ * Look up all supported languages for a specific post
+ *
+ * @param {String} lang The current lang of a post to look up.
+ * @param {EleventyCollectionItem} post The post to look up.
+ * @return {String[][]} The list of supported languages for a specific post
```
Comment (matthiasrohmer) [inline src/site/_filters/language-list.js:34]: Question: is that a valid type?

File: src/site/_filters/language-list.js [hunk 3]
```diff
@@ -15,52 +15,55 @@
...
+  const url = post.url;
+  const languages = getTranslatedUrls(url).filter(
     (langHref) => langHref[0] !== lang,
   );
 
   // Exit early if there are no translations.
-  if (langHrefs.length === 0) {
-    return '';
-  }
+  if (languages.length === 0) return;
```
Comment (matthiasrohmer) [inline src/site/_filters/language-list.js:44]: That essentially is `return undefined` which is not a valid return type according to the JSDoc. This should probably be `return []`.

File: src/site/_filters/language-list.js [hunk 4]
```diff
@@ -15,52 +15,55 @@
...
+ * Look up all supported languages for a specific post
+ *
+ * @param {String} lang The current lang of a post to look up.
+ * @param {EleventyCollectionItem} post The post to look up.
+ * @return {String[][]} The list of supported languages for a specific post
+ */
 
-module.exports = (url, lang) => {
-  const langHrefs = getTranslatedUrls(url).filter(
+const supportedlanguages = (post, lang) => {
```
Comment (matthiasrohmer) [inline src/site/_filters/language-list.js:37]: Nit: camel-case, so `supportedLanguages`.

File: src/site/_includes/partials/language-list.njk [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    {{'i18n.post.available_in' | i18n(locale)}}:
+    {% for language in languages %}
+      {% set lang = language[0] %}
+      {% set langHref = language[1] %}
+      {% set langName = language[2] %}
+
+      <span>
+        <a href={{langHref}} class="post-signpost__link" translate="no" lang={{lang}}>{{langName}}</a>
+        {%- if loop.index !== numberOfLanguages -%}, {%- endif -%}
+        {% if loop.index === (numberOfLanguages - 1) %} and {% endif %}
```
Comment (matthiasrohmer) [inline src/site/_includes/partials/language-list.njk:15]: I think you don't need `numberOfLanguages` here and the loop can be simplified, as you have `loop.first`, `loop.last` and `loop.length` in Nunjucks. See https://mozilla.github.io/nunjucks/templating.html#:~:text=loop.first%3A%20boolean,loop.last

File: src/site/_includes/partials/sign-post.njk [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      <span class="post-signpost__title">
+        {{'i18n.post.appears_in' | i18n(locale)}}:
+      </span>
+
+      {% for pathName in postPaths %}
+        {% set postHref = '/' + paths[pathName].slug + '/' %}
+
+        <a href={{postHref}} class="post-signpost__link">{{paths[pathName].title | i18n(locale)}}</a>
+
+        {% if loop.index !== numberOfPaths %}
```
Comment (matthiasrohmer) [inline src/site/_includes/partials/sign-post.njk:17]: Use `loop.last` instead.

File: src/site/_filters/language-list.js (hunk 2 above)
Comment (mamieorine) [reply src/site/_filters/language-list.js:34]: Yes, this function will return the array of supported languages like:

```
[
  [ 'en', '/i18n/en/webrtc-standard-announcement/', 'English' ],
  [ 'es', '/i18n/es/webrtc-standard-announcement/', 'Español' ],
  [ 'pt', '/i18n/pt/webrtc-standard-announcement/', 'Português' ],
  [ 'ru', '/i18n/ru/webrtc-standard-announcement/', 'Русский' ],
  [ 'zh', '/i18n/zh/webrtc-standard-announcement/', '中文' ],
  [ 'ja', '/i18n/ja/webrtc-standard-announcement/', '日本語' ],
  [ 'ko', '/i18n/ko/webrtc-standard-announcement/', '한국어' ]
]
```

**Review by matthiasrohmer — APPROVED** [review]

**Pull request merged**